app/workers/unretire_subject_worker.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class UnretireSubjectWorker
+  include Sidekiq::Worker
+  attr_reader :workflow_id
+
+  sidekiq_options queue: :high
+
+  def perform(workflow_id, subject_ids)
+    @workflow_id = workflow_id
+    return unless workflow_exists?
```

Comment:
not blocking and feel free to ignore. I'd probably avoid the extracting this to a method (and setting up the attribute reader etc) and i'd inline this method, e.g.
```suggestion
    return unless Workflow.exists?(id: workflow_id)
```